### DISCUSSION TITLE
Some questions about the .prt5 files

### DISCUSSION THREAD

**Discussion by r-broek** [firemodels/fds#12524] [Q&A] [answered]
Dear FDS developers,

After reading through the documentation and older discussions from the Google group, a few things remain unclear to me regarding the particle quantity .prt5 files:

1. My output files are divided into multiple files that follow the convention **CHID_n.prt5**. The **'n'** is supposedly the mesh ordinal number (from https://groups.google.com/g/fds-smv/c/bErXKpBt-p4/m/-uk5rijgAAAJ). Is there any way that I can combine the files or simply produce only one .prt5 file?

2. The reason why I am asking is that I would like to load the data using the provided Matlab script. Unfortunately, the simulation produces 64 of these files, making this a tedious process. Are there any other methods that make this easier? I have found a Python script but it did not seem to work for me and it looks like it also does not allow multiple files as input.

3. If I want to track multiple particle quantities, let's say the 'PARTICLE_TEMPERATURE' and 'PARTICLE_MASS', am I using the correct syntax on the PART line? **QUANTITIES(2) = 'PARTICLE_TEMPERATURE', 'PARTICLE_MASS'/**

Thank you in advance!

**Comment by mcgratta** [comment, marked as answer]
Each MPI process writes its own prt5 file. If each were to write to a single file, that would slow the simulation down considerably. 

I suggest you write your own Matlab or python parser if you cannot find another. 

QUANTITIES='X', 'Y' should work

**Comment by TristanHehnen** [comment]
I believe that the `fdsreader` can read the particle files directly into Numpy arrays:
https://firedynamics.github.io/fdsreader/part.html


https://github.com/FireDynamics/fdsreader

**Comment by johodges** [comment]
You can also import the data using pyfdstools (https://github.com/johodges/pyfdstools). Once you have it installed there is a single routine to read the data:

import pyfdstools as fds
particle_data, times_data = fds.importParticles(resultDir, chid)

**Comment by r-broek** [comment]
Thank you all for the suggestions!
